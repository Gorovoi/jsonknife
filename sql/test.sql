create extension jsonknife;

SELECT knife_extract('{"a":{"b": {"c": 5.3}}}', '["a","b","c"]'); 
SELECT knife_extract('{"a":{"b": {"c": 5.3}}}', '["a",{"d": "ups"},"c"]'); 
