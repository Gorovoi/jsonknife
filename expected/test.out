create extension jsonknife;
SELECT knife_extract('{"a":{"b": {"c": 5.3}}}', '["a","b","c"]'); 
INFO:  reduce array ["a", "b", "c"]
INFO:  item a
INFO:  item b
INFO:  item c
  knife_extract  
-----------------
 ["a", "b", "c"]
(1 row)

SELECT knife_extract('{"a":{"b": {"c": 5.3}}}', '["a",{"d": "ups"},"c"]'); 
INFO:  reduce array ["a", {"d": "ups"}, "c"]
INFO:  item a
INFO:  item {"d": "ups"}
INFO:  item c
      knife_extract       
--------------------------
 ["a", {"d": "ups"}, "c"]
(1 row)

